lib/stdlib/src/erl_tar.erl

```diff
@@ -1253,37 +1259,38 @@ parse_numeric(<<First, _/binary>> = Bin) ->
 
 parse_octal(Bin) when is_binary(Bin) ->
     %% skip leading/trailing zero bytes and spaces
-    do_parse_octal(Bin, <<>>).
-do_parse_octal(<<>>, <<>>) ->
-    0;
+    do_parse_octal(Bin, 0).
+
 do_parse_octal(<<>>, Acc) ->
-    case io_lib:fread("~8u", binary:bin_to_list(Acc)) of
-        {error, _} -> throw({error, invalid_tar_checksum});
-        {ok, [Octal], []} -> Octal;
-        {ok, _, _} -> throw({error, invalid_tar_checksum})
-    end;
-do_parse_octal(<<$\s,Rest/binary>>, Acc) ->
+    Acc;
+do_parse_octal(<<$\s, Rest/binary>>, Acc) ->
     do_parse_octal(Rest, Acc);
 do_parse_octal(<<0, Rest/binary>>, Acc) ->
     do_parse_octal(Rest, Acc);
 do_parse_octal(<<C, Rest/binary>>, Acc) ->
-    do_parse_octal(Rest, <<Acc/binary, C>>).
+    Digit = C - $0,
+    if
```

Comment:
would the following be faster?

Rather than using guard functions, `band` to limit the range to `0-7` then match that the digit hasn't changed.

e.g
```erlang
case Digit band 7 of
    Digit ->
        do_parse_octal(Rest, Acc bsl 3 bor Digit);
    _ ->
        throw({error, invalid_tar_checksum})
end
```

Reply:
That would likely be slightly faster. Changed.